### PR TITLE
CSDK-1037 reorganize repo

### PR DESCRIPTION
**Pull request by aconchillo**

### CODE REVIEW

**Comment by aconchillo** [comment]
@jpgneves We now have a new default branch `daily-3.4.0` and this is the first PR, which is basically a rebase of our old branch `3.4.0-daily-5304` on top of the new one. From now one we should create PRs instead of pushing to main. I believe I have protected it the branch this way.

Once we merge this, I will create a submodule update in `daily-x`.

File: deps/catch/single_include/catch2/catch.hpp [hunk 1]
```diff
@@ -10732,7 +10732,9 @@ namespace Catch {
 
     // 32kb for the alternate stack seems to be sufficient. However, this value
     // is experimentally determined, so that's not guaranteed.
-    static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
+    // NOTE(aleix): MINSIGSTKSZ is not a constant expresion so compilation
+    // fails. Commeting out for now.
```
Comment (jpgneves) [inline deps/catch/single_include/catch2/catch.hpp]: I think I've seen this issue before, and it was totally an environment thing (LLVM pre-14, maybe?). May be worth taking a second look.

**Review by jpgneves — APPROVED** [review]
LGTM, but see note. :)

File: deps/catch/single_include/catch2/catch.hpp (hunk 1 above)
Comment (aconchillo) [reply deps/catch/single_include/catch2/catch.hpp]: Oh, yes. I reverted the change. But I added a few fixes on catch.hpp (copied form mainstrean) to be able to build tests on M1.

Comment (aconchillo) [reply deps/catch/single_include/catch2/catch.hpp]: Actually, the `MINSIGSTKSZ` change is necessary to be able to build tests (only on Linux though). I have investigated and this was due to a change in glibc. I've checked `catch2` upstream and they don't use that code anymore.

In any case, for now I have removed my change.

**Pull request merged**